packages/experimental/Badge/src/Badge.styling.ts

```diff
@@ -70,12 +69,16 @@ export const stylingSettings: UseStylingOptions<BadgeProps, BadgeSlotProps, Badg
       ['iconSize', 'iconColor', 'color'],
     ),
     text: buildProps(
-      (tokens: BadgeTokens, theme: Theme) => ({
-        ...fontStyles.from(tokens, theme),
-        color: tokens.textColor || tokens.color,
-        paddingHorizontal: tokens.textPadding,
-      }),
-      ['color', 'textColor', 'textPadding', ...fontStyles.keys],
+      (tokens: BadgeTokens, theme: Theme) => {
+        return {
+          style: {
+            ...fontStyles.from(tokens, theme),
+            color: tokens.color,
+            ...getIconMargin(tokens),
```

Comment:
Is getIconMargin to supposed to be fed into text? If so should it be renamed to getTextMargin?

Reply:
yeah, I'll rename. I named it like that because we add margin when there's an icon :)